### PR TITLE
WINC-945: [hack] Automate gitlab version updates

### PR DESCRIPTION
**Pull request by wgahnagl**
Enables gitlab version updates to be performed from the pre-release script.

### CODE REVIEW

File: hack/pre-release.sh [hunk 1]
```diff
@@ -303,6 +318,92 @@ dist_git_update() {
...
+  cd $temp
+  clone_repo "https://gitlab.cee.redhat.com/cpaas-products/openshift-winc.git"
+
+  GITLAB_LOCATION=$temp/openshift-winc
+  cd $GITLAB_LOCATION
+
+  latest_hash=$(git rev-parse HEAD)
+
+  gitlab_initial_branch=$(git branch --show-current)
+  new_branch=$(create_git_branch_list $stream-stream-bump $gitlab_initial_branch)
```
Comment (aravindhp) [inline hack/pre-release.sh:349]: Is this branch name unique? What happens if this is not deleted after the MR merges?

File: hack/pre-release.sh [hunk 2]
```diff
@@ -303,6 +318,92 @@ dist_git_update() {
...
+
+  git mv versions/release-$wmco_version versions/release-$updated_version
+  rm versions/release-$updated_version/advisory_map.yml
+
+  # Example: matches "Windows Containers 8.0.1"
+  sed -i "s/Windows Containers $wmco_version/Windows Containers $updated_version/" versions/release-$updated_version/release.yml
+
+  commit_message="Update version to $updated_version
+
+  This commit was generated by hack/pre-release.sh"
```
Comment (aravindhp) [inline hack/pre-release.sh:360]: Does this actually create a well formed commit message? I suspect there will be extra whitespaces added which are not needed.

File: hack/pre-release.sh [hunk 3]
```diff
@@ -303,6 +318,92 @@ dist_git_update() {
...
+
+  # Example: matches version="8.0.1"
+  sed -i "s/version=\"$wmco_version\"/version=\"$updated_version\"/" $dockerfile_location
+
+  # Example: matches commit: d0be5e812c3e3914d3cf3c464623e688673565fb
+  sed -i "s/commit: [0-9a-f]{40}/commit: $latest_hash/"  upstream_sources.yml
+
+  commit_message="[$base_branch] Update version to $updated_version
+
+  This commit was generated by hack/pre-release.sh"
```
Comment (aravindhp) [inline hack/pre-release.sh:393]: Same question here.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @wgahnagl. Please address my comments.

File: hack/pre-release.sh (hunk 1 above)
Comment (wgahnagl) [reply hack/pre-release.sh:349]: the [switch_git_branch ](https://github.com/openshift/windows-machine-config-operator/blob/713a399a6d085721458c1d602a287ce4edf8cf4c/hack/common.sh#L85) function in common.sh that I use [here](https://github.com/openshift/windows-machine-config-operator/blob/713a399a6d085721458c1d602a287ce4edf8cf4c/hack/pre-release.sh#L350) deletes the branch if there's a duplicate. This is the same function used in the submodule update script and elsewhere in this file.

File: hack/pre-release.sh [hunk 4]
```diff
@@ -352,10 +454,12 @@ git submodule update --init --recursive  > /dev/null 2>&1
 WMCO_VERSION=$(sed -n -e 's/^.*WMCO_VERSION ?= //p' Makefile)
 updated_version=$(increment_version $WMCO_VERSION $stream)
 
-if [[ $DISTGIT == "false" ]]; then
-  github_update $base_branch $stream $WMCO_VERSION $updated_version
-else
+if [[ $DISTGIT == "true" ]]; then
```
Comment (aravindhp) [inline hack/pre-release.sh:457]: Nit: I suggest making this a case statement.

**Review by aravindhp — APPROVED** [review]

File: hack/pre-release.sh [hunk 5]
```diff
@@ -303,6 +318,92 @@ dist_git_update() {
...
+
+  git mv versions/release-$wmco_version versions/release-$updated_version
+  rm versions/release-$updated_version/advisory_map.yml
+
+  # Example: matches "Windows Containers 8.0.1"
+  sed -i "s/Windows Containers $wmco_version/Windows Containers $updated_version/" versions/release-$updated_version/release.yml
+
+  commit_message="Update version to $updated_version
+
+This commit was generated by hack/pre-release.sh"
```
Comment (alinaryan) [inline hack/pre-release.sh:360]: Make this in-line with the other lines?

**Review by alinaryan — COMMENTED** [review]
Thanks for working on this,

File: hack/pre-release.sh (hunk 5 above)
Comment (wgahnagl) [reply hack/pre-release.sh:360]: if I do that, the commit message is improperly indented 😔

Comment (saifshaikh48) [inline hack/pre-release.sh:360]: Is it possible to add the command executed to the commit message? something like 
> "Ran `./pre-release.sh master x -d`"

incorporating `$@` which holds all arguments passed into the shell script could be sufficient

Comment (wgahnagl) [reply hack/pre-release.sh:360]: that would also include the arguments passed along to the function as well, which is the old version and the updated one. Would that be OK?

Comment (saifshaikh48) [reply hack/pre-release.sh:360]: no worries, this was more of a "nice to have" anyway

File: hack/pre-release.sh [hunk 6]
```diff
@@ -303,6 +318,92 @@ dist_git_update() {
...
+# takes 2 arguments
+# 1st arg: stream to update. Can be x, y or z to represent the x y or z stream.
+# 2nd arg: current WMCO version
+# 3rd arg: updated WMCO version
+gitlab_update() {
+ if [ "$#" -ne 3 ]; then
+    echo incorrect parameter count for dist_git_update $#
+    return 1
+  fi
+  stream=$1
```
Comment (saifshaikh48) [inline hack/pre-release.sh:331]: is this arg used? I would expect it to be -- are the the gitlab steps for x stream vs y/z all the same?

Comment (wgahnagl) [reply hack/pre-release.sh:331]: I use it for naming the branch, but I could remove it if we want

Comment (saifshaikh48) [reply hack/pre-release.sh:328]: I see, we should leave it them. What I was trying to imply here was that it'll be used to see if we have to `Change CDN repo number to correct one for release binary` (x stream only, which will be handled later in the story you're creating)

File: hack/pre-release.sh [hunk 7]
```diff
@@ -263,9 +278,6 @@ dist_git_update() {
 
   previous_version=$(grep -Po 'previous_version="\K.*(?=")' render_templates)
 
-  # Example: matches version="v8.0.1"
-  sed -i "s/version=\"v[0-9]\+\(\.[0-9]\+\)\+\"/version=\"v$updated_version\"/g" Dockerfile
-
```
[the remark this answers is not in the document]
Comment (saifshaikh48) [reply hack/pre-release.sh:268]: > and why it's being done.

because render_templates does this for us...

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Pull request merged**